azure_models.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"go.ytsaurus.tech/yt/go/yson"
+)
+
+type AzureUser struct {
+	// PrincipalName is unique human-readable Azure user field, used (possibly with changes)
+	// for the corresponding YTsaurus user's `name` attribute.
+	PrincipalName string `yson:"principal_name"`
+
+	AzureID     ObjectID `yson:"id"`
+	Email       string   `yson:"email"`
+	FirstName   string   `yson:"first_name"`
+	LastName    string   `yson:"last_name"`
+	DisplayName string   `yson:"display_name"`
+}
+
+func NewAzureUser(attributes map[string]any) (*AzureUser, error) {
+	bytes, err := yson.Marshal(attributes)
+	if err != nil {
+		return nil, err
+	}
+
+	var azureUser AzureUser
+	err = yson.Unmarshal(bytes, &azureUser)
+	if err != nil {
+		return nil, err
+	}
+	return &azureUser, nil
```

Comment:
This marshall/unmarshall trick is cool, but it may give misleading feeling that AzureUser struct somehow related to yson encoding/decoding. But I think json and any other encoder can be here.

I suppose this small lib https://pkg.go.dev/github.com/mitchellh/mapstructure#example-Decode can help to replace yson marshal/unmarshal boilerplate here and in GetRaw() and same code in Groups and in future ldap source code.

But I not insist on changing this code.